### DISCUSSION TITLE
Submitting pyside6 package to AppImageHub

### DISCUSSION THREAD

**Discussion by dynobo** [beeware/briefcase#691] [Q&A] [answered]
Hi,

I've build a package with PySide6, and want to submit the AppImage to https://appimage.github.io/apps.
But their [onboarding tests](https://github.com/AppImage/appimage.github.io/runs/5724586749?check_suite_focus=true) fail on my briefcase-generated AppImage with the error:

```
from PySide6 import QtCore, QtWidgets
ImportError: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.28' not found 
(required by /run/firejail/appimage/.appimage-9142/usr/app_packages/PySide6
/Qt/lib/libQt6Core.so.6)
```

The maintainer of the Hub suggested, that the lib was [build on a too new system](https://github.com/AppImage/appimage.github.io/pull/2853#issuecomment-1080942115), but my understanding was that briefcase builds the AppImage within an Ubuntu 16.04 docker container, which should not be too new.

I don't know enough to figure out, what/where the root cause of this problem is.
Could someone please help me out here?

Thanks!

**Comment by freakboy3742** [comment, marked as answer]
Briefcase uses an Ubuntu 18.04 base Docker image (we used 16.04 in the past, but that was updated over a year ago). 

From my testing, it appears to be an issue with PySide6 itself. I've got a an Ubuntu 18.04 test machine, and I get the same GlibC error - which would be consistent with the fact that Ubuntu 18.04 ships with GlibC 2.27. 

Based on [this post](https://www.qt.io/blog/qt6-development-hosts-and-targets), it would appear that Qt may not support Ubuntu 18.04 - however, I'm not sufficiently familiar with Qt to be able to say that canonically. I suspect this may be an issue you need to take up with Qt support channels. 

If Qt doesn't support 18.04, then you'll need to develop a custom deployment for a different base image. To do this, you'll need to:
1. Compile the [Python-linux-support](https://github.com/beeware/Python-linux-support/) project on a more recent Linux (it looks like Ubuntu 20.04 should work), generating a support package tarball.

2. Add a `support_package` instruction to your `pyproject.toml` that points at that support tarball

3. Fork the [briefcase-linux-appimage-template](https://github.com/beeware/briefcase-linux-appimage-template) and update the Dockerfile to use the same base image version that you used to compile the support package

4. Add a `template` instruction to your `pyproject.tomml` that points at your template fork.

That should then allow you to build an AppImage that is based on a Linux distro that has Glibc 2.28, and will thus support Pyside6.

At some point in the future, Briefcase will likely do this upgrade in an official capacity - but I can't say when that will happen.

In the meantime, I've logged #692 to track the problem.
